File: SDialogs/src/main/java/smith/lib/alerts/dialog/callbacks/OnInputClickCallBack.java
package smith.lib.alerts.dialog.callbacks;

/**
 * On input data callback for InputSDialog.
 */
@FunctionalInterface
public interface OnInputClickCallBack {
    /**
     * triggered when clicking positive button in InputSDialog.
     *
     * @param input return the inputted data as string.
     */
    void onClick(String input);
}

File: SDialogs/src/main/java/smith/lib/alerts/dialog/callbacks/OnInputClickCallBack.kt
package smith.lib.alerts.dialog.callbacks

/**
 * On input data callback for InputSDialog.
 */
interface OnInputClickCallBack {
    /**
     * triggered when clicking positive button in InputSDialog.
     * @param input return the inputted data as string.
     */
    fun onClick(input: String)
}
